File: BEE 1040.cpp
#include<iostream>
#include<iomanip>
#define FIXED_FLOAT(x) std::fixed << std::setprecision(1) << (x)

using namespace std;

int main(){
    float n1, n2, n3, n4, score;
    cin >> n1 >> n2 >> n3 >> n4;

    float avg = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / (2 + 3 + 4 + 1);

    cout << "Media: " << FIXED_FLOAT(avg) << endl;

    if (avg >= 7){
        cout << "Aluno aprovado." << endl;
    }
    else if (avg < 5){
        cout << "Aluno reprovado." << endl;
    }
    else{
        cout << "Aluno em exame." << endl;
        cin >> score;
        cout << "Nota do exame: " << FIXED_FLOAT(score) << endl;
        float new_avg = (avg + score) / 2;
        if (new_avg >= 5){
            cout << "Aluno aprovado." << endl;
        }
        else{
            cout << "Aluno reprovado." << endl;
        }
        cout << "Media final: " << FIXED_FLOAT(new_avg) << endl;
    }
    return 0;
}

File: BEE 1040.java
import java.io.IOException;
import java.util.Scanner;

public class Main{
    public static void main(String[] args) throws IOException{
        Scanner input = new Scanner(System.in);
        float n1 = input.nextFloat();
        float n2 = input.nextFloat();
        float n3 = input.nextFloat();
        float n4 = input.nextFloat();

        float avg = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / (2 + 3 + 4 + 1);

        System.out.printf("Media: %.1f\n", avg);

        if (avg >= 7){
            System.out.printf("Aluno aprovado.\n");
        }
        else if (avg < 5){
            System.out.printf("Aluno reprovado.\n");
        }
        else{
            System.out.printf("Aluno em exame.\n");
            float score = input.nextFloat();
            System.out.printf("Nota do exame: %.1f\n", score);
            float new_avg = (avg + score) / 2;
            if (new_avg >= 5){
                System.out.printf("Aluno aprovado.\n");
            }
            else{
                System.out.printf("Aluno reprovado.\n");
            }
            System.out.printf("Media final: %.1f\n", new_avg);
        }
    }
}